### PR TITLE
PAL-1086-data-reader-reactive-streams

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: s3-resource-service/src/test/java/uk/gov/gchq/palisade/service/resource/s3/util/S3ResourceBuilderTest.java [hunk 1]
```diff
@@ -29,18 +31,44 @@
...
+
+        // Given
+        var directoryResourceUri = "s3://bucket/some/";
+
+        // When
+        var directoryResource = AbstractResourceBuilder.create(directoryResourceUri);
+
+        // Then
+        assertThat(directoryResource)
+                .as("Check that when building a resource with an s3 prefix, it is an instance of a S3Resource")
```
Comment (pd104923) [inline s3-resource-service/src/test/java/uk/gov/gchq/palisade/service/resource/s3/util/S3ResourceBuilderTest.java:55]:
```suggestion
                .as("Check that when building a resource with an s3 prefix, it is an instance of a DirectoryResource")
```

File: s3-resource-service/src/test/java/uk/gov/gchq/palisade/service/resource/s3/util/S3ResourceBuilderTest.java [hunk 2]
```diff
@@ -29,18 +31,44 @@
...
+
+        // Given
+        var bucketResourceUri = "s3://bucket/";
+
+        // When
+        var bucketResource = AbstractResourceBuilder.create(bucketResourceUri);
+
+        // Then
+        assertThat(bucketResource)
+                .as("Check that when building a resource with an s3 prefix, it is an instance of a S3Resource")
```
Comment (pd104923) [inline s3-resource-service/src/test/java/uk/gov/gchq/palisade/service/resource/s3/util/S3ResourceBuilderTest.java:68]:
```suggestion
                .as("Check that when building a resource with an s3 prefix, it is an instance of a SystemResource")
```

**Review by pd104923 — CHANGES_REQUESTED** [review]
A few questions and a couple of suggestions

File: NOTICES.md [hunk 3]
```diff
@@ -6,17 +6,21 @@ List of third-party dependencies grouped by their license type
 * Gson ([com.google.code.gson:gson:2.8.6](https://github.com/google/gson/gson))
 * akka-stream-alpakka-s3 ([com.lightbend.akka:akka-stream-alpakka-s3_2.13:3.0.0](https://doc.akka.io/docs/alpakka/current))
 * akka-stream ([com.typesafe.akka:akka-stream_2.13:2.6.10](https://akka.io/))
+* Apache Avro ([org.apache.avro:avro:1.8.2](http://avro.apache.org))
 * Apache Hadoop Common ([org.apache.hadoop:hadoop-common:3.2.1](no url defined))
 * AssertJ fluent assertions ([org.assertj:assertj-core:3.18.0](https://assertj.github.io/doc/assertj-core/))
+* AssertJ fluent assertions ([org.assertj:assertj-core:3.19.0](https://assertj.github.io/doc/assertj-core/))
```
Comment (developer01189998819991197253) [inline NOTICES.md:12]: we dont need both versions of assertJ

File: avro-serialiser/pom.xml [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    <properties>
+        <!-- Dependency Versions -->
+        <avro.version>1.8.2</avro.version>
+
+        <!-- Test Dependency Versions -->
+        <akka.version>2.6.10</akka.version>
+        <scala.version>2.13</scala.version>
+        <junit.jupiter.version>5.7.0</junit.jupiter.version>
+        <assertj.version>3.19.0</assertj.version>
```
Comment (developer01189998819991197253) [inline avro-serialiser/pom.xml:61]: change to 3.18.0 like used elsewhere

File: hadoop-data-reader/src/test/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopDataReaderTest.java [hunk 5]
```diff
@@ -56,89 +47,14 @@ void testReadTextFileWithNoRules() throws IOException {
...
-        response.getWriter().write(os);
-        final Stream<String> lines = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(os.toByteArray()))).lines();
-        assertThat(lines.collect(Collectors.toList()))
-                .as("Check the expected values are returned")
-                .asList()
-                .containsOnly("some data", "some more data");
-    }
-
-    @Test
-    void testReadTextFileWithRules() throws IOException {
```
Comment (developer01189998819991197253) [inline hadoop-data-reader/src/test/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopDataReaderTest.java:83]: Do we not want to test with and without rules?

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

**Review by pd104923 — DISMISSED** [review]

File: hadoop-data-reader/src/test/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopDataReaderTest.java (hunk 5 above)
Comment (dev930018) [reply hadoop-data-reader/src/test/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopDataReaderTest.java:83]: Data Readers now have nothing to do with rules, just reading data

**Review by pd104923 — APPROVED** [review]

**Review by developer01189998819991197253 — DISMISSED** [review]

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**